src/docs/arch-transaction.dox

```diff
@@ -1,8 +1,202 @@
+operation, the timestamps associated with the operations in the same transaction don't have to
+be the same as long as they are monotonically increasing.
+
+Apart from the operation level timestamps, the users are also responsible for managing the
+global level timestamps, i.e, the oldest timestamp, and the stable timestamp. The oldest
+timestamp is the timestamp that should be visible by all concurrent transactions. The stable
+timestamp is the minimum timestamp that a new operation can commit at.
+
+Only transactions running in snapshot isolation can run with timestamps.
+
+@subsection Visibility
+
+The visibility of the transactions in WiredTiger considers both the operations' transaction ids
+and timestamps. The operation is visible only when both its transaction id and its timestamp
+are visible to the reading transaction.
+
+To read a key, WiredTiger first traverses all the updates of that key still in memory until a
+visible update is found. The in-memory updates in WiredTiger are organized as a singly linked
+list with the newest update at the head, called the update chain. If no value is visible on the
+update chain, it checks the version on the disk image, which is the version that was chosen to
+be written to disk in the last reconciliation. If it is still invisible, WiredTiger will search
+the history store to check if there is a version visible to the reader there.
+
+The repeated read guarantee under snapshot isolation may break in one case if the timestamps
+committed to the updates are out of order, e.g,
+
+`U@20 -> U@30 -> U@15`
+
+In the above example, reading with timestamp 15 doesn't guarantee to return the third update.
+In some cases, users may read the second update U@30 if it is moved to the history store.
+
+@subsection Durability
+
+WiredTiger transactions support commit level durability and checkpoint level durability. An
+operation is commit level durable if logging is enabled on the table (@ref arch-logging). After
+it has been successfully committed, the operation is guaranteed to survive restart. An operation
+will only survive across restart under checkpoint durability if it is included in the last
+successful checkpoint.
+
+@section Prepared Transactions
+
+WiredTiger introduces prepared transactions to meet the needs of implementing distributed
+transactions through two-phase commit. Prepared transactions only work under snapshot isolation.
+
+Instead of just having the beginning, operating, and rollback or commit phase, it has a
+prepared phase before the rollback or commit phase. After prepare is called, WiredTiger
+releases the transaction's snapshot and prohibits any more read or write operations on the
+transaction.
+
+By introducing the prepared stage, a two-phase distributed transaction algorithm can rely on
+the prepared state to reach consensus among all the nodes for committing.
+
+Along with the prepared phase, WiredTiger introduces the prepared timestamp and durable
```

Comment:
Seeing as we're talking about the timestamps here can we define the guarantee wiredtiger provides with regards to the durable timestamp?